relay-profiling/src/sample.rs

```diff
@@ -226,6 +226,52 @@ impl SampleProfile {
         self.profile
             .strip_pointer_authentication_code(&self.platform, &self.device.architecture);
     }
+
+    fn remove_idle_samples_at_the_edge(&mut self) {
+        self.remove_idle_samples_at_the_start();
+        self.profile.samples.reverse();
+        self.remove_idle_samples_at_the_start();
+        self.profile.samples.reverse();
```

Comment:
How many samples are usually in a profile? Instead of reversing the vector in place, could you make this work with a reverse iterator? Unfortunately there is no `retain_reverse` method.

Reply:
There's a sample every 10ms and I'd say the average duration of a profile is less than 1s. Max duration is 30s. So we could have an average of 100 and max 3000 samples per thread. Usually we see 10 threads in a `cocoa` profile for example. `python`, `rust` and `node` don't have as many though.

I'm trying another approach by finding the position and then using `drain`. Maybe you'll like that one better?